ceasiompy/ActuatorDisk/func/optimalprop.py

```diff
@@ -0,0 +1,368 @@
+    file.write("#\n")
+    file.write("# Author: Ettore Saetta, Lorenzo Russo, Renato Tognaccini.\n")
+    file.write("# Theoretical and Applied Aerodynamic Research Group (TAARG),\n")
+    file.write("# University of Naples Federico II\n")
+    file.write(
+        "# ---------------------------------------------------------------------------------\n"
+    )
+    file.write("#\n")
+    file.write("MARKER_ACTDISK= \n")
+    file.write("CENTER= \n")
+    file.write("AXIS= \n")
+    file.write("RADIUS= " + str(radius) + "\n")
+    file.write("ADV_RATIO= " + str(advanced_ratio) + "\n")
+    file.write("NROW= " + str(stations) + "\n")
+    file.write("# rs=r/R        dCT/drs       dCP/drs       dCR/drs\n")
+    for i in range(0, stations):
+        file.write(f"  {r[i]:.7f}     {CTrs[i]:.7f}     {CPrs[i]:.7f}     0.0\n")
+    file.close()
+
+    log.info("Warning: present version requires input in SI units.")
+
+
+def thrust_calculator(
+    stations,
+    total_thrust_coefficient,
+    radius,
+    hub_radius,
+    advanced_ratio,
+    free_stream_velocity,
+    prandtl,
+    blades_number,
+):
+
+    dstations = float(stations)
+
+    # Resize the vectors using the number of radial stations.
+    r = np.empty(stations)
+    non_dimensional_radius = np.empty(stations)
+    dCp = np.empty(stations)
+    induced_velocity_distribution = np.empty(stations)
+    new_axial_interference_factor = np.empty(stations)
+    old_axial_interference_factor = np.empty(stations)
+    initial_axial_interference_factor = np.empty(stations)
+    dCt_new = np.empty(stations)
+    dCt_old = np.empty(stations)
+    dCt_0 = np.empty(stations)
+    delta_pressure = np.empty(stations)
+    correction_function = np.empty(stations)
+    optimal_axial_interference_factor = np.empty(stations)
+    optimal_rotational_interference_factor = np.empty(stations)
+    dCt_optimal = np.empty(stations)
+
+    non_dimensional_hub_radius = hub_radius / radius
+
+    # Computation of the non-dimensional radial stations.
+    for i in range(1, stations + 1):
+        r[i - 1] = i / dstations
+        if r[i - 1] <= non_dimensional_hub_radius:
+            i_hub = i - 1
+
+    diameter = 2 * radius
+    n = free_stream_velocity / (diameter * advanced_ratio)
+    omega = n * 2 * pi
+
+    # Computation of the tip loss Prandtl correction function
+    if prandtl is True:
+        for i in range(0, stations):
+            correction_function[i] = (2 / pi) * acos(
+                exp(
+                    -0.5
+                    * blades_number
+                    * (1 - r[i])
+                    * sqrt(1 + (omega * radius / free_stream_velocity) ** 2)
+                )
+            )
+
+    else:
+        for i in range(0, stations):
+            correction_function[i] = 1.0
+
+    # Computation of the non-dimensional radius
+    for i in range(0, stations):
+        non_dimensional_radius[i] = omega * r[i] * radius / free_stream_velocity
```

Comment:
You can probably fill numpy arrays without for loops, it will be shorter and faster